server/python/server.py

```diff
@@ -34,20 +39,38 @@ def onboard_user():
         business_type='individual',
         country='US'
     )
+    # Store the account ID.
+    session['account_id'] = account.id
 
-    return_url = request.headers['origin']
-    
+    origin = request.headers['origin']
+    account_link_url = _generate_account_link(account.id, origin)
+    try:
+        return jsonify({'url': account_link_url})
+    except Exception as e:
+        return jsonify(error=str(e)), 403
+
+
+@app.route('/onboard-user/refresh', methods=['GET'])
+def onboard_user_refresh():
+    if 'account_id' not in session:
+        return redirect('/')
+
+    account_id = session['account_id']
+
+    origin = ('https://' if request.is_secure else 'http://') + request.headers['host']
+    account_link_url = _generate_account_link(account_id, origin)
+    return redirect(account_link_url)
+
+
+def _generate_account_link(account_id, origin):
     account_link = stripe.AccountLink.create(
         type='onboarding',
-        account = account.id,
-        failure_url = '{}/failure.html'.format(return_url),
-        success_url = '{}/success.html'.format(return_url)
+        account=account_id,
+        failure_url=f'{origin}/onboard-user/refresh',
```

Comment:
Couldn't this end up in an infinite loop if someone goes wrong during the on-boarding process?

Reply:
No. The error conditions that can cause the user to end up at `failure_url` also cause the account link create call to 400.